jax/_src/numpy/fft.py

```diff
@@ -86,7 +86,9 @@ def _fft_core(func_name, fft_type, a, s, axes, norm):
         s += [max(0, 2 * (a.shape[axes[-1]] - 1))]
     else:
       s = [a.shape[axis] for axis in axes]
-  transformed = lax.fft(a, fft_type, tuple(s)) * _fft_norm(jnp.array(s), func_name, norm)
+  norm_dtype = jnp.result_type(0., a.real.dtype)
```

Comment:
This is not exactly the same as the type promotion used within `lax.fft`... perhaps it would be simpler to first compute `lax.fft` and then use this dtype to compute `_fft_norm` rather than doing the type promotion in two places?

Reply:
The normalization factor is always real though. Do you think it's worth converting it to complex value and do complex multiplications (for most of the cases)?

Reply:
Actually, I don't know if (complex x real) and (complex x complex) have the same cost

Reply:
To be clear, what I meant is something like this:
```python
transformed = lax.fft(a, fft_type, tuple(s))
transformed *= _fft_norm(jnp.array(s, dtype=transformed.real.dtype), func_name, norm)
```

Reply:
I see. That's much cleaner. I have changed it.